pandas/core/internals/blocks.py

```diff
@@ -1975,6 +1976,18 @@ def external_values(self):
         return self.values.astype(object)
 
 
+class PeriodExtensionBlock(ObjectValuesExtensionBlock):
+    """
+    Used by PeriodArray to ensure proper type conversions
+    """
+
+    def _can_hold_element(self, element: Any) -> bool:
```

Comment:
this looks good, but is there any way to do this within ObjectValuesExtensionBlock?  (big picture we're looking to get less special-casing of Block subclasses)

so maybe in ObjectValuesExtensionBlock could write

```
def _can_hold_element(self, element):
        tipo = maybe_infer_dtype_type(element)
        if tipo is not None:
            return issubclass(tipo.type, self.dtype.type)
        return isinstance(element, self.dtype.type)
```

?  Might need to check/test that this works for IntervalDtype too (wouldnt surprise me if we see the analogous bug there)